### DISCUSSION TITLE
SUMO Simpla configuration loading

### DISCUSSION THREAD

**Discussion by CiscoFrisco** [eclipse/mosaic#108] [Q&A]
Hello all,

I'm currently working on a project using SUMO that incorporates platooning vehicles. For that purpose, I decided to use Simpla to have vehicles behave as platoons. 

The problem is that from what I saw in Simpla's webpage from SUMO's documentation ([here](https://sumo.dlr.de/docs/Simpla.html)), to load a simpla configuration they make use of a python module.

Is there a way to load such configuration to MOSAIC so that vehicles use the simpla configuration when a SUMO scenario is executed.

Best regards,
Francisco Filipe

**Comment by kschrab** [comment]
Hi, as far as I understand, Simpla itself is a python module. MOSAIC, however, just starts the SUMO executable and connects to its TraCI server using a socket connection. I never tested it, but it might be possible to start a custom sh/bat-script instead of the SUMO executable. That script could then execute the python-script you want including simpla. I also think that SUMO already provides the possibility to let multiple TraCI clients to connect to it.

The correct location to use the python script instead of the SUMO executable would be [here](https://github.com/eclipse/mosaic/blob/7b268cbc25cea18b8db76c56bcc68c97f142e830/fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/AbstractSumoAmbassador.java#L253)

But keep in mind, this is just a theoretical approach I've never tried out.

**Reply by CiscoFrisco** [reply]
What about implementing an ambassador that communicates via socket with the python script? Would that work or would that have any synchronization problems during the simulation? In theory, the ambassador would only need to load the config to the sumo interface advancing time with TraCI would already take into account the platooning configuration.
